src/Symfony/Component/Console/Application.php

```diff
@@ -148,13 +148,8 @@ public function run(InputInterface $input = null, OutputInterface $output = null
 
             $renderException($e);
 
-            $exitCode = $e->getCode();
-            if (is_numeric($exitCode)) {
-                $exitCode = (int) $exitCode;
-                if (0 === $exitCode) {
-                    $exitCode = 1;
-                }
-            } else {
+            $exitCode = (int) $e->getCode();
+            if (! is_numeric($e->getCode()) || 0 === $exitCode) {
```

Comment:
Are you sure that this will cause no bug, i'm not.

Reply:
Well, if the test isn't trusted, I am not sure, but I think it was, at least this they must be